pkg/controller/utils/metadata/shared_metadata.go

```diff
@@ -101,84 +116,59 @@ func (sm *SharedMetadata) GetOrCreateClusterUID() (string, error) {
 	return sm.clusterUID, nil
 }
 
-func (sm *SharedMetadata) setupFromOperator() error {
-	sm.clusterName = os.Getenv(constants.DDClusterName)
-
-	if sm.credsManager == nil {
-		return fmt.Errorf("credentials Manager is undefined")
-	}
-
-	creds, err := sm.credsManager.GetCredentials()
-	if err != nil {
-		return err
+func (sm *SharedMetadata) GetOrCreateClusterName() string {
+	if sm.clusterName != "" {
+		return sm.clusterName
 	}
 
-	// API key
-	sm.apiKey = creds.APIKey
-
-	return nil
-}
-
-func (sm *SharedMetadata) setupFromDDA(dda *v2alpha1.DatadogAgent) error {
+	// Set cluster name - try operator first, then DDA
+	// TODO: not ideal really; maybe we could drop cluster name from metadata or extract it as part of rest of metadata instead of tieing with credentials
+	sm.clusterName = os.Getenv(constants.DDClusterName)
 	if sm.clusterName == "" {
-		if dda.Spec.Global != nil && dda.Spec.Global.ClusterName != nil {
+		// Fallback to DDA cluster name
+		dda, err := sm.getDatadogAgent()
+		if err == nil && dda.Spec.Global != nil && dda.Spec.Global.ClusterName != nil {
 			sm.clusterName = *dda.Spec.Global.ClusterName
 		}
 	}
-
-	if sm.apiKey == "" {
-		apiKey, err := sm.getCredentialsFromDDA(dda)
-		if err != nil {
-			return err
-		}
-		sm.apiKey = apiKey
-
-		// if API key is set from DDA, also update request URL if needed
-		if dda.Spec.Global != nil && dda.Spec.Global.Site != nil {
-			mdfURL := url.URL{
-				Scheme: defaultURLScheme,
-				Host:   defaultURLHostPrefix + *dda.Spec.Global.Site,
-				Path:   defaultURLPath,
-			}
-			sm.requestURL = mdfURL.String()
-		}
-	}
-
-	return nil
+	return sm.clusterName
 }
 
-// setCredentials attempts to set up credentials and cluster name from the operator configuration first.
-// If cluster name is empty (even when credentials are successfully retrieved from operator),
-// it falls back to setting up from DatadogAgent to ensure we have a valid cluster name.
-func (sm *SharedMetadata) setCredentials() error {
-	err := sm.setupFromOperator()
-	if err == nil && sm.clusterName != "" {
-		return nil
+// getApiKeyAndURL retrieves the API key and request URL from the operator or DDA
+// and sets the cluster name from the operator or DDA in the SharedMetadata struct
+func (sm *SharedMetadata) getApiKeyAndURL() (*string, *string, error) {
+	// Get credentials (operator → DDA fallback handled internally), dda can be nil
+	dda, _ := sm.getDatadogAgent()
```

Comment:
If there's an issue with DDA retrieval here, should we log the error and note that creds will only be set via the operator? Since the error is discarded, we don't have visibility into any errors related to k8s api call or failure to list DDAs.

Reply:
..ideally we can improve `CredentialManager` to handle DDA internally.